### ISSUE TITLE
new pointer design

### ISSUE THREAD

**Issue by s0170071** [jomjol/AI-on-the-edge-device#335]
sigh... why can't I just have a standard pointer on my water meter...

![grafik](https://user-images.githubusercontent.com/33855380/132732484-c8e1a697-cc3e-4623-9fea-668cbaf36921.png)

What would you need to include that type of pointer in your training ? 
Here in this case, the DNN detected the pointer to point to 3.5.

**Comment by jomjol** [comment]
I would need about 100 pictures for each value. Here the correct value would be: "6.0". The result needs to be stored in the image name with the first 3 charecters. Here the naming would be: ``3.5_XXXXX.jpg`` where XXXXX can be whatever you want.

If you only have about 60-70% of the images I would give it a try.

**Comment by s0170071** [comment]
I look into it. Can I use the ESP camera like above or does the reflection bother you ? 
Also I noticed that intermediate values are problematic. E.g.
![grafik](https://user-images.githubusercontent.com/33855380/132765268-081244b1-bb3c-4299-8996-60f146ec3a06.png)
unfortunately that digit is the litre counter so it often sits at intermediate positions.

**Comment by jomjol** [comment]
Reflections could be a topic, but if it's only for you, then this will work, as your reflection is alway on the same spot.

Intermediate values should be recognized as "N". Therefore I only need the full values. Only if some of the intermediate values are detected wrong (not "N"), it makes sense to update my "N"-examples as well.

**Comment by s0170071** [comment]
Since my meter does even count litres with a digit, I almost always get at least one N.
I did now split it into two numbers, making the last digit a separate reading and add it within OpenHAB if a result comes along. 

How is the flow rate limit operating?  It is a comma separated (float?) amount that must not be  exceeded between readings. Does this apply to the raw result or after shifting the comma ? Would be nice if one could set this independently for all numbers.
![grafik](https://user-images.githubusercontent.com/33855380/132859537-1b0c83e5-ba9b-48bf-a3b6-89c0092c7188.png)

**Comment by jomjol** [comment]
It applies to the shifted value, but it is not really a flow rate, as the time is not considered, only the difference between the last two values.

**Comment by s0170071** [comment]
can I change the values independent from each other in the ini file ? Current file reads: 

```

[PostProcessing]
main.DecimalShift = -2
;last.DecimalShift = 
PreValueUse = true; <============
PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 0.1
;last.MaxRateValue = 
ErrorMessage = false
CheckDigitIncreaseConsistency = true

```


If I appended the measurement prefix, can I have separate settings for both ? E.g. 


```

[PostProcessing]
main.DecimalShift = -2
;last.DecimalShift = 
main.PreValueUse = true ; <============
last.PreValueUse = false ; <============

PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 0.1
;last.MaxRateValue = 
ErrorMessage = false
CheckDigitIncreaseConsistency = true

```

If I have a single rolling digit, I do not want any consistency check. But if I have all the digits before, that makes sense.

**Comment by jomjol** [comment]
No, the usage of the prevalue is currently switch on for all or for none. Currently this setting cannot be applied individually to dedicated numbers.

**Comment by s0170071** [comment]
If any of the digits is not recognized, nothing or the old value will be pushed to MQTT, right ? 
Would it be possible to treat the last digit as 0 if it is not recognized ? In my case it only works sometimes since it often shows an intermediate value. 

So setting it to zero would not harm because the strictly rising rule would filter it.  But I would get results from the other digits more often.

**Comment by jomjol** [comment]
Why don't you use the option PreValue. In this case you will get the last correct found number and not just a zero.

I'm currently working on an option to also detect the intermediate states for the digits (e.g. 5.5 in the picture above).

**Comment by s0170071** [comment]
Well, the effect that Im seeing is this: 
I got two readings defined, one with 4 digits, the other one with 5 digits. The later one includes the last digit, which is often not readable. MQTT shows: 
```
4-dig       5- dig
37.32      37.321
37.33
           37.321
37.34
           37.321
37.35
           37.321
37.36
           37.365
          

```
     
Does that explain the effect ? The last digit prevents correct readout. 
As a workaround I do use both in openHAB and merge the numbers there. The larger one is obviously the correct one.
But in terms of compute efficiency this is not so good. Treating the last digit as 0 if it was not readable could solve this.

**Comment by jomjol** [comment]
Do you have switched off `` CheckDigitIncreaseConsistency``?

**Comment by s0170071** [comment]
CheckDigitIncreaseConsistency is enabled.

**Comment by jomjol** [comment]
Then I would switch that off. It is not recommeded for purely digital counters. If the changes between two measures get too big, this algo will be way off.

**Comment by s0170071** [comment]
> Why don't you use the option PreValue. In this case you will get the last correct found number and not just a zero.
> 
> I'm currently working on an option to also detect the intermediate states for the digits (e.g. 5.5 in the picture above).

Nice, that would also solve the 2 / 7 dilemma.

**Comment by s0170071** [comment]
> Reflections could be a topic, but if it's only for you, then this will work, as your reflection is alway on the same spot.
> 
> Intermediate values should be recognized as "N". Therefore I only need the full values. Only if some of the intermediate values are detected wrong (not "N"), it makes sense to update my "N"-examples as well.

I do have a pretty reliable mis-classification of a  not fully shown "2" as a "7" I think more examples in this direction would be helpful. Should I open a dedicated issue for this ?

**Comment by s0170071** [comment]
Is that fir for the digits or also the analog pointers?


Am 29. Oktober 2021 19:13:23 schrieb jomjol ***@***.***>:

**Comment by jomjol** [comment]
No - the analog took some more iterations - but  now are available, too: https://github.com/jomjol/neural-network-analog-needle-readout/blob/main/ana0900s3_longq.tflite
